fs-api-blockio/test/Main.hs

```diff
@@ -44,12 +42,16 @@ instance Arbitrary ByteString where
   arbitrary = BS.pack <$> arbitrary
   shrink = fmap BS.pack . shrink .  BS.unpack
 
-fromByteString :: PrimMonad m => ByteString -> m (MutableByteArray (PrimState m))
-fromByteString bs = thawByteArray (ByteArray ba) 0 (SBS.length sbs)
-  where !sbs@(SBS.SBS ba) = SBS.toShort bs
+fromByteStringPinned :: PrimMonad m => ByteString -> m (MutableByteArray (PrimState m))
+fromByteStringPinned bs = do
+  mba <- newPinnedByteArray (BS.length bs)
+  forM_ (zip [0..] (BS.unpack bs)) $ \(i, x) -> writeByteArray mba i x
```

Comment:
this could be faster, but perhaps it doesn't matter

Reply:
It's slow but straightforward. The bytestrings the tests are generating are relatively small